### DISCUSSION TITLE
Is there a way to use this plugin within my API controllers?

### DISCUSSION THREAD

**Discussion by selected-pixel-jameson** [ComfortablyCoding/strapi-plugin-transformer#45] [Q&A]
Is there a way to use this plugin within my API controllers? For example I'm hooking into a process when a model is updated and I want to send a request to a third-party service with this data.

**Comment by ComfortablyCoding** [comment]
Yes the transform functionality is just a service of the plugin so it can be used anywhere you have access to the strapi obj.

**Comment by selected-pixel-jameson** [comment]
@ComfortablyCoding Could you provide an example of this? I was having trouble trying to figure out exactly how to do that.

**Comment by ComfortablyCoding** [comment]
It should just be 
```js
// ..
const data = dataFromEvent;
const transforms = {removeAttributesKey: true, removeDataKey: true };
const transformedData  = strapi.services('plugin::transformer.transformService).transformResponse(transforms,data);
// ..

**Reply by briankandersen** [reply]
This is probably a stupid question. Is there any way to reach the plugin service using axios or fetch?

**Reply by ComfortablyCoding** [reply]
You cannot expose services as an endpoint. 

You can create a route with a controller that uses this service, that is convention for strapi.
